barretenberg/cpp/src/barretenberg/dsl/acir_format/acir_format.cpp

```diff
@@ -379,13 +381,17 @@ HonkRecursionConstraintsOutput<Builder> process_honk_recursion_constraints(
     size_t idx = 0;
     std::vector<OpeningClaim<stdlib::grumpkin<Builder>>> nested_ipa_claims;
     std::vector<StdlibProof<Builder>> nested_ipa_proofs;
+    bool is_root_rollup = false;
```

Comment:
this bool checks to see if any of the honk recursion constraints set the proof type to be ROOT ROLLUP HONK and that's how we know whether to run the full IPA recursive verifier or not.